### PR TITLE
[KeyVault] Handle Nullable Parameters for Certificate Auto-Renewal in Set-AzKeyVaultCertificatePolicy

### PR DESCRIPTION
**Pull request by notyashhh**
## Description

This PR addresses issue #25649, where users were unable to disable auto-renewal on a KeyVault certificate policy. The changes include:

- **Manual Validation for Nullable Parameters:** The parameters `RenewAtNumberOfDaysBeforeExpiry` and `RenewAtPercentageLifetime` were originally set with ValidateRange attributes, which caused issues when trying to set them to null. The validation logic has been moved to the `ExecuteCmdlet` method to manually validate these parameters only when they are not null.

- **Updated Command Behavior:** Adjusted the behavior of `Set-AzKeyVaultCertificatePolicy` to allow for proper handling of nullable parameters, ensuring that users can disable auto-renewal by setting these parameters to null.

- **Improved Pipeline Handling:** Addressed an issue where the `Curve`, `KeySize`, and `CertificateTransparency` properties were incorrectly being validated from the pipeline due to `ValueFromPipelineByPropertyNam`e. These validations have been adjusted to prevent incorrect mappings and allow the command to function as expected.

## Mandatory Checklist

- Please choose the target release of Azure PowerShell. (⚠️**Target release** is a different concept from **API readiness**. Please click below links for details.)
  - [x] [General release](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Public preview](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Private preview](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Engineering build](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] No need for a release

- [x] Check this box to confirm: **I have read the [_Submitting Changes_](../blob/main/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/main/CONTRIBUTING.md) and reviewed the following information:**

## Additional Notes:
- This change resolves an issue where the Azure portal did not reflect the actual state of the certificate policy after modifying renewal settings through PowerShell. This ensures consistency across PowerShell and the Azure portal.
- Users can now effectively disable auto-renewal by setting the appropriate parameters to null without encountering validation errors.

This PR improves the usability of the Set-AzKeyVaultCertificatePolicy cmdlet and ensures that it behaves as expected when managing certificate policies within Azure KeyVault.

* **SHOULD** update `ChangeLog.md` file(s) appropriately
    * For SDK-based development mode, update `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`.
        * A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header in the past tense. 
    * For autorest-based development mode, include the changelog in the PR description.
    * Should **not** change `ChangeLog.md` if no new release is required, such as fixing test case only.
* **SHOULD** regenerate markdown help files if there is cmdlet API change. [Instruction](../blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
* **SHOULD** have proper test coverage for changes in pull request.
* **SHOULD NOT** adjust version of module manually in pull request

### CODE REVIEW

File: src/KeyVault/KeyVault/Commands/Certificate/Policy/SetAzureKeyVaultCertificatePolicy.cs [hunk 1]
```diff
@@ -271,6 +272,19 @@ public override void ExecuteCmdlet()
         {
             if (ShouldProcess(Name, Properties.Resources.SetCertificatePolicy))
             {
+
+                // Manually Validate `RenewAtNumberOfDaysBeforeExpiry` and `RenewAtPercentageLifetime`
+                if (RenewAtNumberOfDaysBeforeExpiry.HasValue && (RenewAtNumberOfDaysBeforeExpiry < 1 || RenewAtNumberOfDaysBeforeExpiry > int.MaxValue))
+                {
+                    throw new ArgumentOutOfRangeException(nameof(RenewAtNumberOfDaysBeforeExpiry), "Value must be between 1 and int.MaxValue.");
```
Comment (BethanyZhou) [inline src/KeyVault/KeyVault/Commands/Certificate/Policy/SetAzureKeyVaultCertificatePolicy.cs:279]: throw AzPS Exception, see https://github.com/Azure/azure-powershell-common/tree/main/src/Common/Exceptions

File: src/KeyVault/KeyVault/Commands/Certificate/Policy/SetAzureKeyVaultCertificatePolicy.cs [hunk 2]
```diff
@@ -261,7 +262,7 @@ public class SetAzureKeyVaultCertificatePolicy : KeyVaultCmdletBase
         /// Elliptic Curve Name of the key
         /// </summary>
         [Parameter(Mandatory = false,
-                   ValueFromPipelineByPropertyName = true,
+                   ValueFromPipelineByPropertyName = false,
```
Comment (BethanyZhou) [inline src/KeyVault/KeyVault/Commands/Certificate/Policy/SetAzureKeyVaultCertificatePolicy.cs:265]:
```suggestion
```
Remove this attribute directly

**Review by BethanyZhou — CHANGES_REQUESTED** [review]
check following issues reported by our CIs. If it's false positive, suppress them by https://github.com/Azure/azure-powershell/blob/main/documentation/Debugging-StaticAnalysis-Errors.md#breaking-changes

D:\a\_work\1\s\artifacts/StaticAnalysisResults\BreakingChangeIssues.csv Errors
  "Module","ClassName","Target","Severity","ProblemId","Description","Remediation"
  "Az.KeyVault","Microsoft.Azure.Commands.KeyVault.SetAzureKeyVaultCertificatePolicy","Set-AzKeyVaultCertificatePolicy","0","1050","The parameter set '__AllParameterSets' for cmdlet 'Set-AzKeyVaultCertificatePolicy' has been removed.","Add parameter set '__AllParameterSets' back to cmdlet 'Set-AzKeyVaultCertificatePolicy'."
  "Az.KeyVault","Microsoft.Azure.Commands.KeyVault.SetAzureKeyVaultCertificatePolicy","Set-AzKeyVaultCertificatePolicy","0","1050","The parameter set 'ByValue' for cmdlet 'Set-AzKeyVaultCertificatePolicy' has been removed.","Add parameter set 'ByValue' back to cmdlet 'Set-AzKeyVaultCertificatePolicy'."
  "Az.KeyVault","Microsoft.Azure.Commands.KeyVault.SetAzureKeyVaultCertificatePolicy","Set-AzKeyVaultCertificatePolicy","0","1050","The parameter set 'ExpandedRenewNumber' for cmdlet 'Set-AzKeyVaultCertificatePolicy' has been removed.","Add parameter set 'ExpandedRenewNumber' back to cmdlet 'Set-AzKeyVaultCertificatePolicy'."
  "Az.KeyVault","Microsoft.Azure.Commands.KeyVault.SetAzureKeyVaultCertificatePolicy","Set-AzKeyVaultCertificatePolicy","0","1050","The parameter set 'ExpandedRenewPercentage' for cmdlet 'Set-AzKeyVaultCertificatePolicy' has been removed.","Add parameter set 'ExpandedRenewPercentage' back to cmdlet 'Set-AzKeyVaultCertificatePolicy'."

File: src/KeyVault/KeyVault/Commands/Certificate/Policy/SetAzureKeyVaultCertificatePolicy.cs [hunk 3]
```diff
@@ -271,6 +272,19 @@ public override void ExecuteCmdlet()
         {
             if (ShouldProcess(Name, Properties.Resources.SetCertificatePolicy))
             {
+
```
[the remark this answers is not in the document]
Comment (notyashhh) [reply src/KeyVault/KeyVault/Commands/Certificate/Policy/SetAzureKeyVaultCertificatePolicy.cs:272]: I have created a private helper function in the same class `ValidateArguments`

File: src/KeyVault/KeyVault/Commands/Certificate/Policy/SetAzureKeyVaultCertificatePolicy.cs (hunk 2 above)
Comment (notyashhh) [reply src/KeyVault/KeyVault/Commands/Certificate/Policy/SetAzureKeyVaultCertificatePolicy.cs]: Updated this in the code, removed the instances of `ValueFromPipelineByPropertyName` where it was set to `false`

File: src/KeyVault/KeyVault/Commands/Certificate/Policy/SetAzureKeyVaultCertificatePolicy.cs (hunk 1 above)
Comment (notyashhh) [reply src/KeyVault/KeyVault/Commands/Certificate/Policy/SetAzureKeyVaultCertificatePolicy.cs]: Updated this! Thanks for pointing this out!

File: src/KeyVault/KeyVault/Commands/Certificate/Policy/SetAzureKeyVaultCertificatePolicy.cs [hunk 4]
```diff
@@ -313,5 +314,20 @@ public override void ExecuteCmdlet()
...
             }
         }
+
+
+        private void ValidateArguments(int? RenewAtNumberOfDaysBeforeExpiry, int? RenewAtPercentageLifetime)
+        {
+            // Manually Validate `RenewAtNumberOfDaysBeforeExpiry` and `RenewAtPercentageLifetime`
+            if (RenewAtNumberOfDaysBeforeExpiry.HasValue && (RenewAtNumberOfDaysBeforeExpiry < 1 || RenewAtNumberOfDaysBeforeExpiry > int.MaxValue))
+            {
+                throw new Common.Exceptions.AzPSArgumentOutOfRangeException("Value must be between 1 and int.MaxValue.", nameof(RenewAtNumberOfDaysBeforeExpiry));
```
Comment (BethanyZhou) [inline src/KeyVault/KeyVault/Commands/Certificate/Policy/SetAzureKeyVaultCertificatePolicy.cs:324]: Move constant strings to src\KeyVault\KeyVault\Properties\Resources.resx

**Review by BethanyZhou — CHANGES_REQUESTED** [review]
Add Pester tests or Live tests for modified data plane cmdlets

File: src/KeyVault/KeyVault/Commands/Certificate/Policy/SetAzureKeyVaultCertificatePolicy.cs [hunk 5]
```diff
@@ -313,5 +314,20 @@ public override void ExecuteCmdlet()
                 }
             }
         }
+
+
+        private void ValidateArguments(int? RenewAtNumberOfDaysBeforeExpiry, int? RenewAtPercentageLifetime)
```
Comment (BethanyZhou) [inline src/KeyVault/KeyVault/Commands/Certificate/Policy/SetAzureKeyVaultCertificatePolicy.cs:319]:
```suggestion
        private void ValidateArguments()
```
This function should not be designed for validating RenewAtNumberOfDaysBeforeExpiry and RenewAtPercentageLifetime only even though it's for these two parameters for the time being. We can remove them from argument list as these two parameters are available in class private function.

**Review by BethanyZhou — APPROVED** [review]
LGTM

**Pull request merged**